build.ps1

```diff
@@ -257,14 +257,19 @@ function Invoke-Program() {
     [Parameter(Position = 0, Mandatory = $true)]
     [string] $Executable,
     [switch] $OutNull = $false,
+    [switch] $Wait = $false,
     [string] $OutFile = "",
     [Parameter(Position = 1, ValueFromRemainingArguments)]
     [string[]] $Args
   )
 
   if ($ToBatch) {
     # Print the invocation in batch file-compatible format
-    $OutputLine = "`"$Executable`""
+    if ($Wait) {
+      $OutputLine = "start `"$(Split-Path $Executable -leaf)`" /wait `"$Executable`""
```

Comment:
Why the leaf path only?  It seems that we should be able to use the full path right?  I'm concerned about the command not being in the path.

Reply:
There is such thing as "title" in the [parameters](https://learn.microsoft.com/en-us/windows-server/administration/windows-commands/start) of `start` command. And `start` treats first quoted parameter as a title. As we're quoting executable already, it becomes "a title", and then other arguments are treated as `start` options... A 
complete mess. I tried some alternative forms (like quoting whole command string along with options), and that didn't help. Seems like the title could be omitted only if there is no quotes at all.

Basically it is just an arbitrary string, and it could be empty quotes. I just thought it makes sense to provide some readable value, but not too long (i.e. not the full path). Maybe to avoid confusion and excessive code, I just have to use "" here. WDYT about ""?

Reply:
That might work? this seems very confusing 😓

Reply:
Yeah, that actually works.
```
:: ✅ Works, no quotes
> start /wait C:\Windows\System32\msiexec.exe /qn /a a1 TARGETDIR=C:\Downloads\T

:: ✅ Works, the executable is parsed before the quoted argument
> start /wait C:\Windows\System32\msiexec.exe /qn /a "a1" TARGETDIR=C:\Downloads\T

:: 🚫 Not works, quoted part is parsed as title, and then Start looks for a valid option or executable
> start /wait "C:\Windows\System32\msiexec.exe" /qn /a a1 TARGETDIR=C:\Downloads\T
Invalid switch - "/qn".

:: 🚫 Not works. Opens a new cmd instance with the "title" from quoted part
> start /wait "C:\Windows\System32\msiexec.exe /qn /a a1 TARGETDIR=C:\Downloads\T"

:: ✅ Works, as the "title" is found empty
> start /wait "" "C:\Windows\System32\msiexec.exe" /qn /a "a1" TARGETDIR="C:\Downloads\T"
```

This weird parsing behavior is seems to be well known, and is mentioned on [ss64.com](https://ss64.com/nt/start.html) (my favorite source of advanced info and best practices about cmd in particular):
>Always include a TITLE this can be a simple string like "My Script" or just a pair of empty quotes ""
According to the Microsoft documentation, the title is optional, but depending on the other options chosen you can have problems if it is omitted.